span: remove INVERTED check from builder

RaduBerinde opened the pull request:
This check has been carried over since we first added support for JSON
indexes; at that point inverted spans were always single key and thus
had inclusive boundaries.

The check as it stands now is incorrect - there is no reason why we
should always treat an end key as inclusive for an inverted index. In
practice, the only current cases where we might have an exclusive end
key are with geospatial queries, where the spans are not "tight"
anyway (i.e.  false positives are allowed). This might change in the
future though, e.g. if we add support for range queries for JSON
elements.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76865)
<!-- Reviewable:end -->

RaduBerinde commented:
I didn't manage to find a query that results in an exclusive end span (in a real server or in the execbuilder tests).

yuzefovich reviewed (commented):
I'll defer to Becca for approval.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76865#-:-MwT5-EnEn3G-D2AddIq:bd839ug)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RaduBerinde and @rytaft)

___
*[pkg/sql/span/span_builder.go, line 227 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76865#-MwT4cfcDQukrryL_z0y:-MwT4cfcDQukrryL_z0z:b-twc2hm) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6d32a51eaa788ac2575a21e7132be6db97e2a0c/pkg/sql/span/span_builder.go#L227)):*
> ```Go
> // roachpb.Spans and appends them to the provided spans. It appends multiple
> // spans in the case that multiple, non-adjacent column families should be
> // scanned. The forDelete parameter indicates whether these spans will be used
> ```

nit: `forDelete` has been removed recently.
___
*[pkg/sql/span/span_builder.go, line 264 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76865#-MwT4XSi2HHce-kEG77m:-MwT4XSj4-cyeOooi0TO:bk48tj3) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6d32a51eaa788ac2575a21e7132be6db97e2a0c/pkg/sql/span/span_builder.go#L264)):*
> ```Go
> 	}
> 
> 	// We need to advance the end key if it is inclusive or the index is
> ```

nit: the comment needs an update.


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
span: remove INVERTED check from builder

This check has been carried over since we first added support for JSON
indexes; at that point inverted spans were always single key and thus
had inclusive boundaries.

The check as it stands now is incorrect - there is no reason why we
should always treat an end key as inclusive for an inverted index. In
practice, the only current cases where we might have an exclusive end
key are with geospatial queries, where the spans are not "tight"
anyway (i.e.  false positives are allowed). This might change in the
future though, e.g. if we add support for range queries for JSON
elements.

Release note: None

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76865#-:-MwWuRWX-uJDNREX8L1Z:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76865#-:-MwXKw8C4R52kGO7sEmu:b-tlx7ko)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->
